util/crypto/src/secp/generator.rs

```diff
@@ -13,14 +13,21 @@ impl Generator {
     }
 
     pub fn random_privkey(&self) -> Privkey {
-        let mut rng = rand::thread_rng();
-        key::SecretKey::new(&SECP256K1, &mut rng).into()
+        let mut random_slice: [u8; secp256k1::constants::SECRET_KEY_SIZE] = rand::random();
+        loop {
+            match SecretKey::from_slice(&random_slice) {
+                Ok(sec) => return sec.into(),
+                Err(_) => {
+                    random_slice = rand::random();
+                }
+            }
+        }
     }
 
     pub fn random_keypair(self) -> Result<(Privkey, Pubkey), Error> {
-        let mut rng = rand::thread_rng();
-
-        let (sec, publ) = SECP256K1.generate_keypair(&mut rng)?;
+        let random_slice: [u8; secp256k1::constants::SECRET_KEY_SIZE] = rand::random();
+        let sec = SecretKey::from_slice(&random_slice)?;
+        let publ = PublicKey::from_secret_key(&SECP256K1, &sec);
```

Comment:
Use the `random_privkey` above to generate a valid secret_key to maintain the consistency of this function.

https://github.com/nervosnetwork/rust-secp256k1/blob/master/src/lib.rs#L526